src/browser/modules/App/App.tsx

```diff
@@ -83,180 +87,280 @@ import {
-  useEffect(() => {
-    editor.setTheme(derivedTheme)
-  }, [derivedTheme])
+    return rules[this.props.telemetrySettingSource as TelemetrySettingSource]
+  }
 
-  useKeyboardShortcuts(props.bus)
+  eventMetricsCallback = (_: MetricsData) => {}
+  segmentTrackCallback = (_: MetricsData) => {}
+  unsub = () => {}
 
-  const eventMetricsCallback = useRef((_: MetricsData) => {})
-  const segmentTrackCallback = useRef((_: MetricsData) => {})
+  eventHandler = ({ category, label, data: originalData }: MetricsData) => {
+    if (!isRunningE2ETest() && this.shouldSendMetricsData()) {
+      const data = {
+        browserVersion: version,
+        ...originalData
+      }
+      this.eventMetricsCallback({ category, label, data })
+      this.segmentTrackCallback({ category, label, data })
+    }
+  }
 
-  useEffect(() => {
-    const unsub =
-      props.bus &&
-      props.bus.take(
-        METRICS_EVENT,
-        ({ category, label, data: originalData }: MetricsData) => {
-          if (!isRunningE2ETest()) {
-            const data = {
-              browserVersion: version,
-              ...originalData
-            }
-            eventMetricsCallback &&
-              eventMetricsCallback.current &&
-              eventMetricsCallback.current({ category, label, data })
-            segmentTrackCallback &&
-              segmentTrackCallback.current &&
-              segmentTrackCallback.current({ category, label, data })
-          }
-        }
-      )
-    const initAction = udcInit()
-    props.bus && props.bus.send(initAction.type, initAction)
-    return () => unsub && unsub()
-  }, [])
+  componentDidMount() {
+    const { bus } = this.props
+    if (bus) {
+      this.unsub = bus.take(METRICS_EVENT, this.eventHandler)
+      const initAction = udcInit()
+      bus && bus.send(initAction.type, initAction)
+    }
+  }
+  componentWillUnmount() {
+    this.unsub && this.unsub()
+  }
 
-  const {
-    drawer,
-    handleNavClick,
-    activeConnection,
-    connectionState,
-    lastConnectionUpdate,
-    errorMessage,
-    loadExternalScripts,
-    loadSync,
-    syncConsent,
-    browserSyncMetadata,
-    browserSyncConfig,
-    browserSyncAuthStatus,
-    experimentalFeatures,
-    store,
-    codeFontLigatures,
-    defaultConnectionData,
-    useDb,
-    databases
-  } = props
+  render() {
+    const {
+      activeConnection,
+      browserSyncAuthStatus,
+      browserSyncConfig,
+      browserSyncMetadata,
+      bus,
+      codeFontLigatures,
+      connectionState,
+      consentBannerShownCount,
+      databases,
+      defaultConnectionData,
+      drawer,
+      errorMessage,
+      experimentalFeatures,
+      handleNavClick,
+      lastConnectionUpdate,
+      loadExternalScripts,
+      loadSync,
+      openSettingsDrawer,
+      setConsentBannerShownCount,
+      setEnvironmentTheme,
+      store,
+      syncConsent,
+      telemetrySettingSource,
+      themeData,
+      titleString,
+      useDb
+    } = this.props
```

Comment:
Out of scope for this PR, but this number of props here hints that this god-component might be subject for being split into multiple more narrow scoped components.

Reply:
I was thinking the same and left a mental note (it's also missing prop types)